pkg/services/semantics.go

```diff
@@ -28,8 +28,8 @@ type SemanticTokenStruct struct {
 
 var PARAM_REGEX, _ = regexp.Compile(`<<\s*(parameters|pipeline.parameters)\.([A-z0-9-_]*)\s*>>`)
 
-func SemanticTokens(params protocol.SemanticTokensParams, cache *utils.Cache) protocol.SemanticTokens {
-	doc, err := parser.GetParsedYAMLWithCache(params.TextDocument.URI, cache)
+func SemanticTokens(params protocol.SemanticTokensParams, cache *utils.Cache, context *utils.LsContext) protocol.SemanticTokens {
+	doc, err := parser.ParseURIWithCache(params.TextDocument.URI, cache, context)
```

Comment:
`ParseURIWithCache` could mean that we parse an URI (literally the object URI) with a cache, I personally found this name a bit weird

Reply:
What about `ParseFromUriWIthCache` ?

Reply:
(The problem with `GetParsedYAMLWithCache` is that the action described in the name is `get`, like if the function was a simple getter)